File: backend/app/models/eod.py
from datetime import date, datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field, HttpUrl


class ManualReply(BaseModel):
    linkedin_url: HttpUrl
    text: str = Field(min_length=1, max_length=5000)


class ManualBooking(BaseModel):
    linkedin_url: HttpUrl
    meeting_at: datetime


class CofounderEodInput(BaseModel):
    cofounder_id: str
    crs_sent: list[HttpUrl] = Field(default_factory=list)
    dms_sent: list[HttpUrl] = Field(default_factory=list)
    connections_accepted: list[HttpUrl] = Field(default_factory=list)
    replies_received_manual: list[ManualReply] = Field(default_factory=list)
    bookings_manual: list[ManualBooking] = Field(default_factory=list)
    anomalies: list[str] = Field(default_factory=list)
    notes: str = Field(default="", max_length=4000)


class EodSubmitRequest(BaseModel):
    per_cofounder: list[CofounderEodInput]


class EodCofounderPrefill(BaseModel):
    cofounder_id: str
    cofounder_name: str
    shipped: int
    dropped: int
    edited: int
    replies_count: int
    bookings_count: int


class EodPrefillResponse(BaseModel):
    log_date: date
    per_cofounder: list[EodCofounderPrefill]
    last_submitted_at: datetime | None = None


class EodSubmitResponse(BaseModel):
    log_id: str
    nightly_task_id: str

File: frontend/lib/eod.ts
import { api } from "./api";

export type EodCofounderPrefill = {
  cofounder_id: string;
  cofounder_name: string;
  shipped: number;
  dropped: number;
  edited: number;
  replies_count: number;
  bookings_count: number;
};

export type EodPrefill = {
  log_date: string;
  per_cofounder: EodCofounderPrefill[];
  last_submitted_at: string | null;
};

export type EodCofounderInput = {
  cofounder_id: string;
  crs_sent: string[];
  dms_sent: string[];
  connections_accepted: string[];
  replies_received_manual: { linkedin_url: string; text: string }[];
  bookings_manual: { linkedin_url: string; meeting_at: string }[];
  anomalies: string[];
  notes: string;
};

export const eodApi = {
  prefill: () => api.get<EodPrefill>("/api/eod/today"),
  submit: (per_cofounder: EodCofounderInput[]) =>
    api.post<{ log_id: string; nightly_task_id: string }>(
      "/api/eod/submit",
      { per_cofounder },
    ),
};
